traveler-ios-ui/TravelerKitUI/ViewControllers/CancellationViewController.swift

```diff
@@ -15,50 +15,127 @@ protocol CancellationViewControllerDelegate: class {
 }
 
 class CancellationViewController: UITableViewController {
-    @IBOutlet weak var totalRefundLabel: UILabel!
+    @IBOutlet weak var explanationInputTextField: UITextField!
 
     var quote: CancellationQuote?
     weak var delegate: CancellationViewControllerDelegate?
 
+    private var selectedReason: CancellationReason?
+    private var explanation: String?
+    private var selectedReasonIndexPath: IndexPath?
+
     override func viewDidLoad() {
         super.viewDidLoad()
 
-        totalRefundLabel.text = quote?.totalRefund.localizedDescriptionInBaseCurrency
+        explanationInputTextField.delegate = self
     }
 
     // MARK: UITableViewDataSource
 
     override func numberOfSections(in tableView: UITableView) -> Int {
-        return 1
+        return 2
     }
 
     override func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
-        return quote?.products.count ?? 0
+        if section == 0 {
+            return (quote?.products.count ?? 0) + 1
+        } else {
+            return (quote?.cancellationReasons.count ?? 0)
+        }
+    }
+
+    override func tableView(_ tableView: UITableView, titleForHeaderInSection section: Int) -> String? {
+        if section == 1 {
+            return "Please select a reason for cancellation"
+        } else {
+            return nil
+        }
+    }
+
+    override func tableView(_ tableView: UITableView, titleForFooterInSection section: Int) -> String? {
+        if section == 0 {
+            return " "
+        } else {
+            return nil
+        }
     }
 
     override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
-        let cell = tableView.dequeueReusableCell(withIdentifier: "productCancelCell", for: indexPath) as! InfoCell
+        if indexPath.section == 0 {
+            if indexPath.row >= (quote?.products.count ?? 0) {
+                let cell = tableView.dequeueReusableCell(withIdentifier: "productCancelSummaryCell", for: indexPath) as! InfoCell
```

Comment:
"productCancelSummaryCell" is too similar to "productCancellCell". This makes it difficult to know what the cell represents at a first glance. I would rename "productCancellSummaryCell" to "totalRefundCell" since that is what that cell is actually representing.